src/main/java/tv/rewinside/nbtstorage/nbt/NBTTagLongArray.java

```diff
@@ -0,0 +1,87 @@
+package tv.rewinside.nbtstorage.nbt;
+
+import tv.rewinside.nbtstorage.exceptions.NBTLoadException;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class NBTTagLongArray extends NBTBase {
+
+	private long[] data;
+
+	NBTTagLongArray() {
+	}
+
+	public NBTTagLongArray(long[] along) {
+		this.data = along;
+	}
+
+	@Override
+	void write(DataOutput dataoutput) throws IOException {
+		dataoutput.writeInt(this.data.length);
+
+		for (int i = 0; i < this.data.length; ++i) {
+			dataoutput.writeLong(this.data[i]);
+		}
+
+	}
+
+	@Override
+	void load(DataInput datainput, int i, NBTReadLimiter nbtreadlimiter) throws IOException {
+		int j = datainput.readInt();
+		if (j >= 1 << 24) throw new NBTLoadException("Error while loading LongArray");
+
+		nbtreadlimiter.allocate((long) (64 * j));
+		this.data = new long[j];
+
+		for (int k = 0; k < j; ++k) {
+			this.data[k] = datainput.readLong();
+		}
+
+	}
+
+	@Override
+	public NBTType getType() {
+		return NBTType.LONG_ARRAY;
+	}
+	
+	@Override
+	public long[] getData() {
+		return this.data;
+	}
+
+	@Override
+	public NBTBase clone() {
+		long[] along = new long[this.data.length];
```

Comment:
Rename this to something else. A long variable shouldn't include its type in the name itself